src/actions.js

```diff
@@ -128,6 +128,31 @@ export const fetchWithCreds = (opts) => {
   );
 };
 
+export const fetchWithCredsAndTimeout = (opts, timeoutInMS) => {
+  let didTimeOut = false;
+
+  return new Promise(((resolve, reject) => {
+    const timeout = setTimeout(() => {
+      didTimeOut = true;
+      reject(new Error('Request timed out'));
+    }, timeoutInMS);
+
+    fetchWithCreds(opts).then((response) => {
+      // Clear the timeout as cleanup
+      clearTimeout(timeout);
+      if (!didTimeOut) {
+        resolve(response);
+      }
+    })
+      .catch((err) => {
+        // Rejection already happened with setTimeout
+        if (didTimeOut) return;
+        // Reject with error
+        reject(err);
+      });
+  }));
+};
```

Comment:
i copied this function verbatim from the NDE https://github.com/uc-cdis/data-ecosystem-portal/blob/master/src/actions.js#L125

Reply:
so dont worry too much about reading it =P